lib/Doctrine/ORM/ORMInvalidArgumentException.php

```diff
@@ -195,11 +195,7 @@ public static function invalidIdentifierBindingEntity()
      */
     public static function invalidAssociation(ClassMetadata $targetClass, $assoc, $actualValue)
     {
-        $expectedType = 'Doctrine\Common\Collections\Collection|array';
-
-        if (($assoc['type'] & ClassMetadata::TO_ONE) > 0) {
```

Comment:
Is there any explanation on why we had this code there (commit msg)? Maybe @Ocramius can give us some context

Reply:
No details in the commit message of https://github.com/doctrine/doctrine2/commit/71a6a88de82d23232cdf5377a44804a5f55d29b8 and not sure where I could find details on DDC-3490.